core/src/main/java/org/apache/calcite/util/format/FormatElementEnum.java

```diff
@@ -57,8 +59,17 @@ public enum FormatElementEnum implements FormatElement {
   },
   DAY("EEEE", "The full weekday name") {
     @Override public void format(StringBuilder sb, Date date) {
-      final Work work = Work.get();
-      sb.append(work.eeeeFormat.format(date));
+      final Calendar calendar = Work.get().calendar;
+      calendar.setTime(date);
+      // The Calendar and SimpleDateFormatter do not seem to give correct results
+      // for the day of the week prior to the Julian to Gregorian date change.
+      // So we resort to using a LocalDate representation.
+      LocalDate ld =
```

Comment:
Can use java.sql.Date directly?
`((java.sql.Date) date).toLocalDate()`

Reply:
I didn't know this trick, I will try it.

Reply:
it doesn't work, the subclass relationship goes the other way. sql.date extends util.date.
The calendar-based conversion seems to do the right thing and it's not too complicated.